bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/RegExpFactory.java

```diff
@@ -183,6 +183,11 @@ private static RegExpAtom translateLiteralCharOrEscape(RegExpLiteralCharOrEscape
         if ("&".equals(value)) {
             return createLiteralCharOrEscape("\\&");
         }
+        if (value.startsWith("\\u{") && value.endsWith("}")) {
+            int codePoint = Integer.parseInt(value.substring(3, value.length() - 1), 16);
```

Comment:
Let's do below instead
```ballerina
return createLiteralCharOrEscape(Utils.unescapeBallerina(value));
```

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/40368/commits/e26daea0c1d5b4a0c11d068f104eda74dd5ce5c5.